### ISSUE TITLE
Implement an optional more manual device/interface selection.

### ISSUE THREAD

**Issue by hmatuschek** [hmatuschek/qdmr#435]
As frequently requested and needed for some very cheap devices (DR-1801), a proper interface and device selection mechanism is needed. The majority of the work is already present. 

There are a few things to be done:

- [x] Enable interface selection dialog.
- [x] Enable device selection dialog.
- [x] Allow for disabling the auto-detection.
- [x] Interface specification for CLI.
- [x] Update documentation. Also explain auto-detection mechanism.

**Comment by hmatuschek** [comment]
Enabled device selection for DR-1801UV, also have a look at the settings dialog. There, you can disable the auto-detection, if you want to.

**Comment by DH1PK** [comment]
Hi!
I built qdmr-435-implement-manual-device-interface-selection.

In settings I deactivated auto-detect.
Then clicked "download codeplug".

Debug in src/application.cc@1001: Set icon theme to 'light'.
Debug in src/application.cc@81: Search for translation in '/usr/local/share/locale/de_DE/LC_MESSAGES'.
Debug in src/application.cc@81: Search for translation in '/usr/local/share/locale/de/LC_MESSAGES'.
Debug in src/application.cc@84: Installed translator for locale 'de_DE'.
Debug in src/repeaterbookcompleter.cc@328: Loaded repeater cache of 0 entries.
Debug in lib/userdatabase.cc@132: Loaded user database with 259618 entries from /home/pk/.local/share/DM3MAT/qdmr/user.json.
Debug in lib/talkgroupdatabase.cc@140: Loaded talk group database with 1675 entries from /home/pk/.local/share/DM3MAT/qdmr/talkgroups.json.
serialnmea: No known GPS device found. Specify the COM port via QT_NMEA_SERIAL_PORT.
Debug in src/application.cc@137: Last known position: 
Debug in src/application.cc@189: Create main window using icon theme 'light'.
serialnmea: No known GPS device found. Specify the COM port via QT_NMEA_SERIAL_PORT.
qt.positioning.geoclue2: Already stopped
Debug in src/application.cc@554: Last device is invalid, search for new one.
Debug in lib/usbserial.cc@118: Search for serial port with matching VID:PID 28e9:18a.
Debug in lib/usbserial.cc@118: Search for serial port with matching VID:PID 1fc9:94.
Debug in lib/hid_libusb.cc@128: Search for HID interfaces matching VID:PID 15a2:73.
Debug in lib/dfu_libusb.cc@157: Search for DFU devices matching VID:PID 483:df11.
Debug in lib/c7000device.cc@236: Search for C7000 devices matching VID:PID 1206:227.
Debug in src/application.cc@558: No save device found, continue searching for unsave ones.
Debug in lib/usbserial.cc@118: Search for serial port with matching VID:PID 28e9:18a.
Debug in lib/usbserial.cc@118: Search for serial port with matching VID:PID 1fc9:94.
Debug in lib/hid_libusb.cc@128: Search for HID interfaces matching VID:PID 15a2:73.
Debug in lib/dfu_libusb.cc@157: Search for DFU devices matching VID:PID 483:df11.
Debug in lib/usbserial.cc@118: Search for serial port with matching VID:PID 67b:23a3.
Debug in lib/c7000device.cc@236: Search for C7000 devices matching VID:PID 1206:227.
ERROR in src/application.cc@562: Kein passendes Gerät gefunden.

I see no way to select the tty or to select DR1801UV. Maybe I am looking in the wrong places?
Should I select DR1801UV before pressing download? Or should I be asked after pressing download?
And switching off auto-detection doesnt seem to change anything for me.

I tried two different cables, both with the same result.
Both cables report: idVendor=1a86, idProduct=7523

And I found my second DR1801UV which has a working codeplug installed, no difference.

Maybe the zip-file is not yet updated?
md5sum qdmr-435-implement-manual-device-interface-selection.zip
537806a85dc9b21566ef380c24a8a7c5  qdmr-435-implement-manual-device-interface-selection.zip

Regards,
Patrick
